lib/test_ids/allocator.rb

```diff
@@ -14,6 +14,72 @@ def initialize(configuration)
       @config = configuration
     end
 
+    # Allocates a softbin number from the range specified in the test flow
+    # It also keeps a track of the last softbin assigned out from a particular range
+    # and uses that to increment the pointers accordingly.
+    # If a numeric number is passed to the softbin, it uses that number.
+    # The configuration for the TestId plugin needs to pass in the bin number and the options from the test flow
+    # For this method to work as intended.
+    def next_in_range(softbin, options)
+      unless options.nil?
```

Comment:
I would expect this method to be dealing with returning either a next in line number, or else calling a reclaim method